File: leetcode/medium/311-sparse-matrix-multiplication/311-sparse-matrix-multiplication.kt
class Solution {
    fun multiply(mat1: Array<IntArray>, mat2: Array<IntArray>): Array<IntArray> {
        val result = Array(mat1.size) { IntArray(mat2[0].size) { 0 } }

        for (i in mat1.indices) {
            for (j in mat2[0].indices) {
                var sum = 0

                for (k in mat2.indices) {
                    sum += mat1[i][k] * mat2[k][j]
                }

                result[i][j] = sum
            }
        }

        return result
    }
}

File: leetcode/medium/311-sparse-matrix-multiplication/311-sparse-matrix-multiplication.py
from typing import List


class Solution:
    def multiply(self, mat1: List[List[int]], mat2: List[List[int]]) -> List[List[int]]:
        result = []

        for i in range(len(mat1)):
            row = []

            for j in range(len(mat2[0])):
                value = 0

                for k in range(len(mat2)):
                    value += mat1[i][k] * mat2[k][j]

                row.append(value)

            result.append(row)

        return result
